src/ctap/storage.rs

```diff
@@ -285,6 +319,11 @@ impl PersistentStore {
             .count()
     }
 
+    #[cfg(feature = "with_ctap2_1")]
+    pub fn remaining_credentials(&self) -> usize {
+        MAX_SUPPORTED_RESIDENTIAL_KEYS - self.count_credentials()
```

Comment:
We might want to use `MAX_SUPPORTED_RESIDENTIAL_KEYS.checked_sub(self.count_credentials()).ok_or(XXX)` at some point. But for that we need to update all usages of the persistent storage to support failure. I'll do that in a separate PR once we know which error code we want to return for the different kinds of errors.

Reply:
Fixed after the rebase (more like rework).